ndhistogram/src/axis/cyclic.rs

```diff
@@ -0,0 +1,199 @@
+use super::{Axis, BinInterval, Uniform};
+use std::fmt::Debug; // TODO Display
+
+use num_traits::{Float, Num, NumCast, NumOps};
+use serde::{Deserialize, Serialize};
+
+/// A wrap-around axis with equal-sized bins.
+///
+/// An axis with `N` equally-spaced, equal-sized bins, in `[low, high)`.
+/// Entries outside this interval get wrapped around.
+/// There are no overflow bins so this axis has exactly `N` bins.
+///
+/// # Examples
+/// 1D histogram with 4 bins distributed around a circle.
+/// ```
+/// use ndhistogram::{ndhistogram, Histogram};
+/// use ndhistogram::axis::{Axis, BinInterval, Cyclic};
+/// let mut hist = ndhistogram!(Cyclic::new(4, 0.0, 360.0));
+/// hist.fill(& 45.0         ); // Add entry at 45 degrees
+/// hist.fill(&(45.0 + 360.0)); // Add entry at 45 degrees + one whole turn
+/// hist.fill(&(45.0 - 360.0)); // Add entry at 45 degrees + one whole turn backwards
+/// // All 3 above entries end up in the same bin
+/// assert_eq!(hist.value(&45.0), Some(&3.0));
+/// // Lookup also wraps around
+/// assert_eq!(hist.value(&(45.0 + 360.0)), Some(&3.0));
+/// assert_eq!(hist.value(&(45.0 - 360.0)), Some(&3.0));
+/// ```
+/// Time of day
+/// ```
+/// use ndhistogram::{ndhistogram, Histogram};
+/// use ndhistogram::axis::{Axis, BinInterval, Cyclic};
+/// let bins_per_day = 24;
+/// let hours_per_bin = 1;
+/// let start_at_zero = 0;
+/// let four_pm = 16;
+/// let mut hist = ndhistogram!(Cyclic::with_step_size(
+///     bins_per_day, start_at_zero, hours_per_bin
+/// ));
+/// hist.fill(&40);                               // The 40th hour of the week ...
+/// assert_eq!(hist.value(&four_pm), Some(&1.0)); // ... is at 4 pm.
+/// ````
+#[derive(Default, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Serialize, Deserialize)]
+pub struct Cyclic<T = f64> {
+    axis: Uniform<T>,
+}
+
+impl<T> Cyclic<T>
+where
+    T: PartialOrd + Num + NumCast + NumOps + Copy,
+{
+    /// Create a wrap-around axis with `nbins` uniformly-spaced bins in the range `[low, high)`.
+    ///
+    /// Only implemented for [Float]. Use [Cyclic::with_step_size] for integers.
+    ///
+    /// # Panics
+    /// Panics under the same conditions as [Uniform::new].
+    pub fn new(nbins: usize, low: T, high: T) -> Self
+    where
+        T: Float,
+    {
+        Self {
+            axis: Uniform::new(nbins, low, high),
+        }
+    }
+
+    /// Create a wrap-around axis with `nbins` uniformly-spaced bins in the range `[low, low+num*step)`.
+    /// # Panics
+    /// Panics under the same conditions as [Uniform::new].
+    pub fn with_step_size(nbins: usize, low: T, step: T) -> Self {
+        Self {
+            axis: Uniform::with_step_size(nbins, low, step),
+        }
+    }
+}
+
+impl<T> Cyclic<T> {
+    /// Low edge of axis (excluding wrap-around) // TODO or should this be - infinity?
```

Comment:
> // TODO or should this be - infinity?

Haha. Yes this is a good question. I think what you have done already is the best thing to do.
The user may need access to finite `low()` and `high()` for eg plotting. Returning `inf` in this case isn't very useful.